app/src/main/java/com/github/se/eventradar/model/repository/location/NominatimLocationRepository.kt

```diff
@@ -0,0 +1,38 @@
+package com.github.se.eventradar.model.repository.location
+
+import com.github.se.eventradar.model.Location
+import com.github.se.eventradar.model.Resource
+import java.io.IOException
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import org.json.JSONArray
+
+class NominatimLocationRepository(
+    val client: OkHttpClient = OkHttpClient(),
+    val requestBuilder: Request.Builder = Request.Builder(),
+    private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO
+) : ILocationRepository {
+  override suspend fun fetchLocation(locationName: String): Resource<Location> {
+    return try {
+      val url = "https://nominatim.openstreetmap.org/search?q=${locationName}&format=json"
+      val request = requestBuilder.url(url).build()
+      val response = withContext(ioDispatcher) { client.newCall(request).execute() }
+      if (!response.isSuccessful) {
+        throw IOException("Request to get location data failed due to IOException ${response}")
```

Comment:
Instead of throwing an exception in the try block, perhaps you can return a failure here?

Reply:
The exception gets caught as it is in the try catch, so it will still return a resource Failure in the end, but I can change it too for better clarity 👍